contents/projectManagement/testDrivenDevelopment.md

```diff
@@ -0,0 +1,207 @@
+<frontmatter>
+  title: Test-Driven Development
+  header: pagetop.md
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+<div class="website-content">
+
+# Test-Driven Development
+
+Author: [Yash Chowdhary](https://github.com/yash-chowdhary)
+
+## What is Test-driven Development? 
+
+Typically, when you write unit tests, you test out the functionality of code you've already written. Test-driven development (a.k.a TDD) is a software development process that emphasizes the opposite - writing tests before production code. It is a design technique aimed at understanding requirements before writing functional code.
+
+### Development Cycle
+
+Robert Martin describes the "Three Laws of TDD" as - 
+> "First Law: You may not write production code until you have written a failing unit test. 
+>
+> Second Law: You may not write more of a unit test than is sufficient to fail, and not compiling is failing. 
+>
+> Third Law: You may not write more production code than is sufficient to pass the currently failing test.”
+
+Figure 1 below shows the general TDD cycle.
+
+<center>
+<pic src="images/tddSteps.jpg" alt="tddSteps" height="650">
+
+_Figure 1. General TDD cycle._[^2]
+</pic>
+</center>
+
+Kent Beck, creator of Xtreme Programming and advocate of TDD, states that the general development cycle is as follows[^1]:
+
+1. **Add a test**
+
+    In this phase of the cycle, all you need to focus on is writing tests. Understanding the product's/feature's requirements and specifications is crucial at this stage of the development cycle. 
+
+1. **Run the test**
+
+    In this phase, you check if the new test added fails. This is an important step, as it shows that the new test does not pass without requiring new code because the required behavior already exists, and also rules out false positives (i.e. the test passes when it shouldn't). Furthermore, the reason for its failure must be clear. 
```

Comment:
Do you mean to say

"This is an important step, as it shows that the new test does not pass without new code because the required behavior **doesn't yet** exists,"

Reply:
You're right, it is confusing. I can simplify it to 
> "This is an important step as it shows that the new test does not pass without requiring new code, and also rules out..."